assets/src/edit-story/app/media/useMediaReducer/reducer.js

```diff
@@ -62,27 +76,40 @@ function reducer(state, { type, payload }) {
     case types.RESET_FILTERS: {
       return {
         ...state,
+        ...resetOnFilterChange,
         searchTerm: '',
         mediaType: '',
       };
     }
 
     case types.SET_SEARCH_TERM: {
       const { searchTerm } = payload;
+      if (searchTerm === state.searchTerm) return state;
       return {
         ...state,
+        ...resetOnFilterChange,
         searchTerm,
       };
     }
 
     case types.SET_MEDIA_TYPE: {
       const { mediaType } = payload;
+      if (mediaType === state.mediaType) return state;
       return {
         ...state,
+        ...resetOnFilterChange,
```

Comment:
This sounds like a method name, could we rename this to something else? Does this include the default/initial state values basically?

Reply:
This seems to share quite a few values with `INITIAL_STATE`, perhaps we could use sth like
`FILTER_RESET_STATE` and then the `INITIAL_STATE` could consist of `FILTER_RESET_STATE` + the values unique to that only?

Reply:
This should be cleaner: https://github.com/google/web-stories-wp/pull/562/commits/59423a673c935063bf0516dd0ca7cb2b1f2b159d